### ISSUE TITLE
Recurring transactions

### ISSUE THREAD

**Issue by mauxtin** [sebdesign/laravel-viva-payments#9]
I am having the same problem with the [initial issue](https://github.com/VivaPayments/API/issues/54) regarding recurring payments.

[The test](https://github.com/sebdesign/laravel-viva-payments/blob/master/tests/Functional/TransactionTest.php#L44) seems to be marked as skipped so I was wondering whether you managed to find a solution to this. (I am getting the same error as you were getting back then.)
```
Invalid transaction. The transaction 6a617543-2bdf-4fea-be69-10e4b09abda8 does not allow recurring charge

```

**Comment by sebdesign** [comment]
Hello @mauxtin,

I haven't found a solution for this yet. I know there are some issues on https://github.com/VivaPayments/API about this, and I hope this will get resolved soon.

**Comment by mauxtin** [comment]
Hi @sebdesign,

I came back to this now and I noticed the following.
I am creating an order from postman directly to viva and through this package. These two requests should be identical.
```
curl --location --request POST 'https://demo.vivapayments.com/api/orders' \
--header 'Authorization: Basic ZmJmMWUwOTAtNjVkZS00Yzc1LWI1ODAtZDlkOWJjNDk0YWQyOldEdi1WKw==' \
--header 'Content-Type: application/json' \
--header 'Content-Type: text/plain' \
--data-raw '{
      "Email": "customer@domain.com",
      "Phone": "2117604000",
      "FullName": "Customer Name",
      "AllowRecurring": true,
      "IsPreAuth": true,
      "Amount": 1000
  }'
```

```
   $order = app(\Sebdesign\VivaPayments\Order::class);

        $orderCode = $order->create(1000,[
            'FullName'        => 'Customer Name',
            'Email'           => 'customer@domain.com',
            'Phone'           => "2117604000",
            'SourceCode'      => 8362,
            'AllowRecurring'  => true,
            'IsPreAuth'       => true,
        ]);

```


However, when using this package the checkbox to allow future recurring payments is missing.
![image](https://user-images.githubusercontent.com/40521145/79766038-20316b80-8330-11ea-9360-53c5800410a4.png)


Any ideas on what is happening? 😕

**Comment by sebdesign** [comment]
Thanks for reporting this @mauxtin, I will investigating this and get back to you by tomorrow.

**Comment by mauxtin** [comment]
Hi @sebdesign, any chance you managed to figure this out?

**Comment by sebdesign** [comment]
@mauxtin I think I've solved it in v4.2.0. Can you update and check if that works for you? Also I noticed the parameter names are in cameCase instead of PascalCase now in the documentation, but I don't think it really matters.

**Comment by mauxtin** [comment]
Yes, that fixed it! I tried to debug it but didn't think the issue would be something like that... 😃 

Thank you for your help!
